elastic/kibana
# [Uptime] Improve monitor charts query

## Summary

Resolves https://github.com/elastic/kibana/issues/29843.

This change would update the query used for fetching monitor chart data, and clean up computations done on the client that probably belong in the server.

## Testing this PR
The changes enacted will have visual results, so aside from providing code review you should be able to see it's working as intended.
- Start up Heartbeat
- Select a monitor
- Verify that the **duration** value in the status bar looks correct. Things to watch for:
  - Make sure the value isn't _really_ fast. We had a bug recently that caused the duration value to be "converted" twice, meaning `val / (1000 * 1000)`. A request to `https://www.elastic.co` is not going to complete in `0.15ms`.
  - Likewise, make sure the value doesn't look too large, i.e. not being converted at all. A request to `https://www.elastic.co` should not take `150,000ms` either.
- Verify the **duration** stats in the duration chart seem like they're on an appropriate scale. I've included a sample screenshot below:
![image](https://user-images.githubusercontent.com/18429259/52649430-549cd280-2eb6-11e9-9d74-bf45b0af4447.png)


~This PR is in progress, and shouldn't be reviewed until https://github.com/elastic/kibana/pull/30441 is merged.~
